GuessTheNumber.java

```diff
@@ -20,5 +21,187 @@
+
+					// Count the digits guessed by players for notifications
+					for (int j = 0; j < theNumberLength; j++) {
+						tmpuserNumberDigit = (int) (tmpuserNumber / Math.pow(10, theNumberLength - 1 - j));
+						tmpuserNumber = (int) (tmpuserNumber % Math.pow(10, theNumberLength - 1 - j));
+
+						// For each entered digit, check the all digits in generated number.
+						tmptheNumber = theNumber;
+						for (int i = 0; i < j + 1; i++) {
+							tmptheNumberDigit = (int) (tmptheNumber / Math.pow(10, theNumberLength - 1 - i));
+							tmptheNumber = (int) (tmptheNumber % Math.pow(10, theNumberLength - 1 - i));
+							// If digits is same, check the position in number.
+							if (tmpuserNumberDigit == tmptheNumberDigit) {
+								if (i == j) {
+									counterSamePosition++;
+								} else {
+									counterDifferentPosition++;
+								}
+							}
+						}
+					}
+					messArr[numberOfSteps] = "Step #" + numberOfSteps + " Number: " + userNumber
+							+ ". You guessed: on the same position:" + counterSamePosition
+							+ ", on the different position:" + counterDifferentPosition;
+
+					System.out.print("\n");
+					for (int i = 1; i <= numberOfSteps; i++) {
+						System.out.println(messArr[i]);
+					}
+				}
+
+				numberOfSteps++;
+
+				// Check user input for same length as generated number.
+				int counterUserNumberLenght = 0;
+				do {
+					System.out.print("Try #:" + numberOfSteps + ". Enter the " + theNumberLength + "-digit number:");
+					userNumber = in.nextInt();
+					// Count the entered length of the number.
+					int tmpun = userNumber;
+					counterUserNumberLenght = 0;
+					while (tmpun > 0) {
+						tmpun = tmpun / 10;
+						counterUserNumberLenght++;
+					}
+				} while (theNumberLength != counterUserNumberLenght);
+
+			} while (theNumber != userNumber);
+
+			// Update highScoreTable for current Player
+			highScoreTable[userId - 1] = numberOfSteps;
+
+			System.out.print(
+					"\nCongratulations, Player number: " + userId + "! You WIN from " + numberOfSteps + " trays!");
+
+			// Prepare player notifications from highScoreTable.
+			System.out.println("\n---- High-Scores Table -----");
+			int topPlayersCounter = 0;
+			for (int i = 0; i < highScoreTable.length; i++) {
+				if (highScoreTable[i] > 0) {
+					System.out.println("User #" + (i + 1) + " score: " + highScoreTable[i] + " trays!");
+				}
+				if (highScoreTable[i] < highScoreTable[userId - 1] && highScoreTable[i] > 0) {
+					topPlayersCounter++;
+				}
+			}
+
+			if (topPlayersCounter > 0) {
+				System.out.print("\nThere are " + topPlayersCounter + " players with better scores!");
+			} else {
+				System.out.print("\nYou are at the top of the table!");
+			}
+
+			// Before the next game, the top 3 scores (which are greater than 0) will be printed
+			// Copy array for sorting
+			highScoreTableSorted = highScoreTable.clone();
+			int temp;
+			// Bubble sort
+			for (int i = 0; i < highScoreTableSorted.length - 1; i++) {

```

Comment:
👍 for the bubble sort

Reply:
we  still didn't learned recursion quicksort :)

Reply:
:)